src/main/scala/com.snowplowanalytics/forex/ForexClient.scala

```diff
@@ -29,64 +38,55 @@ import com.twitter.util.SynchronizedLruMap
 object ForexClient {
 
   /**
-   * Getter for clients with specified caches(optional)
+   * Creates a client with a cache and sensible default ForexConfig
    */
-  def getClient(config: ForexConfig,
-                clientConfig: ForexClientConfig,
-                nowish: MaybeNowishCache = None,
-                eod: MaybeEodCache       = None): ForexClient =
-    clientConfig match {
-      case oerClientConfig: OerClientConfig =>
-        new OerClient(config, oerClientConfig, nowishCache = nowish, eodCache = eod)
-      case _ => throw NoSuchClientException("This client is not supported by scala-forex currently")
-    }
-}
-
-abstract class ForexClient(config: ForexConfig, nowishCache: MaybeNowishCache = None, eodCache: MaybeEodCache = None) {
+  def getClient[F[_]: Sync](appId: String, accountLevel: AccountType): F[ForexClient[F]] =
+    getClient[F](ForexConfig(appId = appId, accountLevel = accountLevel))
 
-  // Assemble our caches
-  object caches {
+  /**
+   * Getter for clients, creating the caches as defined in the config
+   */
+  def getClient[F[_]: Sync](config: ForexConfig): F[ForexClient[F]] = {
+    val nowishCacheF =
+      if (config.nowishCacheSize > 0)
+        LruMap.create[F, NowishCacheKey, NowishCacheValue](config.nowishCacheSize).map(_.some)
+      else Sync[F].pure(Option.empty[NowishCache[F]])
 
-    // LRU cache for nowish request, with (source currency, target currency) as the key
-    // and (date time, exchange rate) as the value
-    val nowish =
-      if (nowishCache.isDefined) {
-        nowishCache
-      } else if (config.nowishCacheSize > 0) {
-        Some(new SynchronizedLruMap[NowishCacheKey, NowishCacheValue](config.nowishCacheSize))
-      } else {
-        None
-      }
+    val eodCacheF =
+      if (config.eodCacheSize > 0)
+        LruMap.create[F, EodCacheKey, EodCacheValue](config.eodCacheSize).map(_.some)
+      else Sync[F].pure(Option.empty[EodCache[F]])
 
-    // LRU cache for historical request, with (source currency, target currency, date time) as the key
-    // and exchange rate as the value
-    val eod =
-      if (eodCache.isDefined) {
-        eodCache
-      } else if (config.eodCacheSize > 0) {
-        Some(new SynchronizedLruMap[EodCacheKey, EodCacheValue](config.eodCacheSize))
-      } else {
-        None
-      }
+    (nowishCacheF, eodCacheF).mapN {
+      case (nowish, eod) =>
+        new OerClient[F](config, nowishCache = nowish, eodCache = eod)
+    }
   }
 
+  def getClient[F[_]: Sync](config: ForexConfig,
+                            nowishCache: Option[NowishCache[F]],
+                            eodCache: Option[EodCache[F]]): ForexClient[F] =
+    new OerClient[F](config, nowishCache, eodCache)
+}
+
+abstract class ForexClient[F[_]](val config: ForexConfig,
+                                 val nowishCache: Option[NowishCache[F]] = None,
```

Comment:
This is quite minor, but I'd like to get rid of default parameters whenever possible and use overloaded constructors instead. Maybe just a personal preference.